src/commands/show.ts

```diff
@@ -79,4 +78,90 @@ export const show = async (): Promise<void> => {
   console.log("  Total Staked Value (SOL) ", tvlStaked.toLocaleString())
   console.log("  Total Liquidity-Pool (SOL) ", tvlLiquidity.toLocaleString())
   console.log("  TVL (SOL) ", (tvlStaked + tvlLiquidity).toLocaleString())
+
+  if ('list' in options) {
+    const {validatorRecords, capacity:validatorCapacity} = await marinadeState.getValidatorRecords()
+    const {stakeInfos, capacity:stakeCapacity} = await marinadeState.getStakeInfos()
+
+    console.log()
+    console.log("  Validator_manager_authority", state.validatorSystem.managerAuthority.toBase58())
+    console.log(`  Stake list account: ${state.stakeSystem.stakeList.account} with ${state.stakeSystem.stakeList.count}/${stakeCapacity} stakes`)
+    console.log("-----------------")
+    console.log("-- Validators ---")
+    console.log(`  Total staked: ${MarinadeUtils.lamportsToSol(state.validatorSystem.totalActiveBalance)} SOL`)
+    console.log(`  List account: ${state.validatorSystem.validatorList.account} with ${state.validatorSystem.validatorList.count}/${validatorCapacity} validators`)
+    console.log("-------------------------------------------------------------")
+
+    const epochInfo = await marinadeState.epochInfo()
+
+    let countStaked = 0;
+    let totalStaked = new BN(0)
+    let totalStakedFullyActivated = new BN(0)
+
+    validatorRecords
+        .forEach((validator, validatorIndex) => {
+
+      // Find stakeInfo by delegation for current validator and active stakes
+      if (validator.activeBalance.toNumber() <= 0) return
+
+      countStaked++;
+
+      const validatorStakes:MarinadeBorsh.StakeInfo[] = stakeInfos
+          .filter(stakeInfo => stakeInfo.stake.Stake?.stake.delegation)
+          .filter(stakeInfo => stakeInfo.stake.Stake?.stake.delegation.voterPubkey.toBase58() == validator.validatorAccount.toBase58())
+          .filter(stakeInfo => MarinadeUtils.U64_MAX.eq(stakeInfo.stake.Stake?.stake.delegation.deactivationEpoch as BN))
+
+      console.log(`${validatorIndex+1}) Validator ${validator.validatorAccount.toBase58()}`
+          + `, marinade-staked ${MarinadeUtils.lamportsToSol(validator.activeBalance).toFixed(2)} SOL`
+          + `, score-pct: ${(validator.score * 100/ state.validatorSystem.totalValidatorScore).toFixed(4)}%, ${validatorStakes.length} stake-accounts`)
+
+      for (const [index, stakeInfo] of validatorStakes.entries()) {
```

Comment:
Could you please split this long function to a few logical pieces to enable others to orient in this long function a bit more easily? I think removing the contents of this huge `if` and all these `for` loops would be beneficial

Reply:
I've just updated: separate list to function.